### ISSUE TITLE
ggspec: aes_params

### ISSUE THREAD

**Issue by ijlyttle** [vegawidget/ggvega#32]
Hi @haleyjeppson,

We were noticing that if there are no `aes_params` in a layer, that the value is `NULL`. If it's empty, could the value instead be an empty `list()`? 

I think we are using an empty `list()` for scales, when there are none.

In general, if we have something where there can be a number of them, but we have none, could we emit an empty `list()`?

**Comment by wenyuyangpku** [comment]
Hi @ijlyttle and @haleyjeppson ,

Sorry to reopen this issue.

When I build the ggschema, I thought `aes_params` should be an array. But I am wrong.

I think in [iris-scatterplot-01.json](https://github.com/vegawidget/ggvega/blob/master/dev/exploratory/iris-scatterplot-01.json), The `aes_params` is something like this:
```json
 "aes_params": {
     "colour": {"value": "red"}
  }
```

And in [01-ggspec-somponents.md](https://github.com/vegawidget/ggvega/blob/master/dev/exploratory/02-ggspec-components.md), `aes_params` is something like this:
```r
##   ..$ aes_params : List of 2
##   .. ..$ shape : num 21
##   .. ..$ fill : chr "white"
```
I think the json format is:
```json
"aes_params": {
    "shape": 21,
    "fill": "white"
 }
```

Both  `aes_params` are not an array. So I think the empty one should not be`[]`. Maybe we can still use `null` or empty object like `{}`.

**Comment by ijlyttle** [comment]
My apologies to both @haleyjeppson and @wenyuyangpku - let me look for some possible solutions...

**Comment by ijlyttle** [comment]
I think what is needed here is an empty *named* list, that we can use for such occasions:

```r
str(jsonlite::fromJSON("{}"))
```

```
Named list()
```

Taking some inspiration from the [BBmisc package](https://github.com/berndbischl/BBmisc/blob/a5a4e45197572b02ddbb86c4fbd96dfc0771666e/R/namedList.R), we could create an internal variable like:

```
empty_named_list <- {
  result <- list()
  setNames(result) <- character(0)

  result
}
```

**Comment by haleyjeppson** [comment]
Hi @ijlyttle and @wenyuyangpku,

I agree that an empty named list is what is need here. I will work on making that change.

**Comment by ijlyttle** [comment]
Fixed in #35
